sdk/trace/span.go

```diff
@@ -241,6 +309,87 @@ func (s *span) SetName(name string) {
 	}
 }
 
+func (s *span) Name() string {
```

Comment:
I'd prefer if the exporters aren't going to have to take and release a whole bunch of Mutexes to read through a ReadOnly type. Can the ReadOnly accessors be simple accessors, and then could the ReadWrite accessors just take a lock and call through to the ReadOnly accessors? It seems that by the time you pass a span to an ordinary exporter, you no longer need the mutex. Is that possible?

Reply:
Thanks for the feedback @jmacd.

When exporting a span we actually lock the mutex just once: The span processor calls `span.Dump()` (which I'll likely rename following the feedback above) which locks the mutex, takes a "snapshot" of the span's state and then releases the mutex.

Example:

https://github.com/johananl/opentelemetry-go/blob/e81b72935d5b1e0c9378906f2c7b2161062af36f/sdk/trace/simple_span_processor.go#L46-L53

The RO accessors are to be used when only a specific span field needs to be read.

Does that make sense to you?

Reply:
Going to resolve this. @jmacd feel free to unresolve if needed.